### ISSUE TITLE
feat: add no-bare-urls rule (MD034)

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#104]
## Overview

Implement the `no-bare-urls` rule (equivalent to markdownlint MD034 / remark-lint `no-literal-urls`).

Closes part of #76.

## Rule Description

URLs must not appear as bare text. They must be wrapped in angle brackets or used inside a proper Markdown link.

**Invalid:**
\`\`\`markdown
Visit https://example.com for more info.
\`\`\`

**Valid:**
\`\`\`markdown
Visit <https://example.com> for more info.
Visit [Example](https://example.com) for more info.
\`\`\`

## Implementation Notes

- Rule key: `no-bare-urls`
- Detect bare `http://` and `https://` URLs not wrapped in `<...>` and not inside a Markdown link `[text](url)` or image `![alt](url)`
- URLs inside fenced code blocks and inline code spans should be ignored
- Config field: `BareUrls bool`

**Comment by shinagawa-web** [comment]
## Implementation Plan

### Rule Behavior

Flag any `http://` or `https://` URL that appears as bare text — i.e., not wrapped in angle brackets and not inside a Markdown link or image syntax.

The following are **valid** and must not be flagged:

| Form | Example |
|---|---|
| Angle bracket URL | `<https://example.com>` |
| Inline link | `[text](https://example.com)` |
| Image | `![alt](https://example.com)` |
| Inside inline code span | `` `https://example.com` `` |
| Inside fenced code block | ignored entirely |

Everything else is a **violation**:

```markdown
Visit https://example.com for details.   ← bare URL
```

---

### Detection Strategy

For each line (outside fenced code blocks):

1. Blank out inline code spans (content between backticks) so URLs inside them are not scanned.
2. Use regex `https?://[^\s<>()\[\]]+` to find all URL occurrences.
3. For each match, inspect the character immediately before the URL:
   - `(` → inside a Markdown link/image → skip
   - `<` → inside an angle bracket URL → skip
   - anything else → bare URL → report violation

Trailing punctuation (`.`, `,`, `)`, `;`, `!`, `?`) is trimmed from the URL before including it in the error message, since it is typically not part of the URL itself.

---

### Edge Cases

| Case | Handling |
|---|---|
| Fenced code block | Skip entirely via existing helpers |
| Inline code span | Blank out before scanning |
| URL at start of line | No preceding character → bare URL |
| Multiple bare URLs on one line | Report each separately |
| Trailing punctuation after URL | Strip from displayed URL in message |
| `http://` and `https://` | Both covered by `https?://` |
| Reference-style links `[text][ref]` | URL not present on the line → not flagged |

---

### Files to Create / Modify

| Action | File |
|---|---|
| Create | `internal/rule/no_bare_urls.go` |
| Create | `internal/rule/no_bare_urls_test.go` |
| Create | `e2e/fixtures/no_bare_urls_valid.md` |
| Create | `e2e/fixtures/no_bare_urls_violation.md` |
| Create | `e2e/config-no-bare-urls.json` |
| Modify | `internal/config/config.go` — add to `Default()` and `DefaultConfigJSON` |
| Modify | `internal/linter/linter.go` — register in `collectErrors()` |
| Modify | `e2e/e2e_test.go` — add E2E test cases, update `DirectoryRecursion` count |

---

### Step-by-Step

1. **Implement `no_bare_urls.go`**
   - Package-level compiled regex: `var bareURLRegex = regexp.MustCompile(`https?://[^\s<>()\[\]]+`)`
   - Helper `stripInlineCode(s string) string` to blank out backtick spans
   - `CheckNoBareURLs(filename string, lines []string, offset int) []LintError`

2. **Create `no_bare_urls_test.go`** (table-driven)
   - Valid: angle bracket URL, inline link, image, inline code, fenced block
   - Invalid: bare `https://`, bare `http://`, multiple on one line, URL at start of line
   - Offset case to verify line number shifting

3. **Update `config.go`**
   - Add `"no-bare-urls": true` to `DefaultConfigJSON`
   - Add entry to `Default()`

4. **Update `linter.go`**
   - Add `IsEnabled("no-bare-urls")` block in `collectErrors()`

5. **Create E2E fixtures**
   - `no_bare_urls_valid.md`: uses `<url>` and `[text](url)` forms only
   - `no_bare_urls_violation.md`: contains at least one bare URL at a known line number
   - `config-no-bare-urls.json`: `"default": false`, only `no-bare-urls` enabled

6. **Update `e2e_test.go`**
   - Add `NoBareURLsValid` / `NoBareURLsViolation` test cases
   - Update `DirectoryRecursion` file count and violation assertions

---

### Error Message

```
no-bare-urls: bare URL found, use angle brackets or a Markdown link: https://example.com
```
